### PR TITLE
identity: Introduce reserved:unmanaged identity

### PR DESCRIPTION
**Pull request by tgraf**
Unmanaged endpoints are currently labeled with the identity reserved:init. This
change was introduced for 1.3. While reserved:init correctly implements the
cluster entity and also allows to define policy. It does not provide clear
visibility and does not allow to define policy for unmanaged pods without also
affecting managed pods in the init phase.

Introduced a new identity reserved:unmanaged to map unmanaged endpoints to
instead. The numeric value for the previous cluster identity can be utilized
for this purpose. There is almost a 1:1 mapping of the previous intent of the
cluster entity so re-using the value will cause the least potential for
disruption when upgrading from 1.2 to 1.3.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5898)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

Legit failure due to typo in header file generation

**Comment by tgraf** [comment]
test-me-please

File: bpf/lib/policy.h [hunk 1]
```diff
@@ -32,14 +32,15 @@
  * - ReservedIdentityWorld	(2)
  *
  * The following identities are given to endpoints so return false for these:
+ * - ReservedIdentityUnmanaged  (3)
  * - ReservedIdentityHealth	(4)
  * - ReservedIdentityInit	(5)
  *
  * Identities 128 and higher are guaranteed to be generated based on user input.
  */
 static inline bool identity_is_reserved(__u32 identity)
```
Comment (jrajahalme) [inline bpf/lib/policy.h:41]: This needs a better name, as also health and unmanaged identities are "reserved"

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
(just to check if there's upgrade/downgrade issues)

**Review by aanm — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

File: pkg/identity/identity_test.go [hunk 2]
```diff
@@ -111,6 +116,15 @@ func (s *IdentityTestSuite) TestAllocateIdentityReserved(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(i.ID, Equals, ReservedIdentityInit)
 	c.Assert(isNew, Equals, false)
+
+	lbls = labels.Labels{
+		labels.IDNameInit: labels.NewLabel(labels.IDNameUnmanaged, "", labels.LabelSourceReserved),
```
Comment (jrajahalme) [inline pkg/identity/identity_test.go]: Why use `IDNameInit` here?

**Review by jrajahalme — APPROVED** [review]
Just small nits

File: bpf/lib/policy.h (hunk 1 above)
Comment (joestringer) [reply bpf/lib/policy.h:41]: Agree, but we can follow this up separately in the effort to improve readability of the BPF code.

**Comment by tgraf** [comment]
green builds. will address feedback from @jrajahalme

**Review by joestringer — APPROVED** [review]
BPF bits LGTM.

File: pkg/identity/identity_test.go (hunk 2 above)
Comment (tgraf) [reply pkg/identity/identity_test.go]: Good catch fixed. It was primarily cosmetic but clearly wrong.

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**